packages/data-mate/src/transforms/field-transform.ts

```diff
@@ -0,0 +1,448 @@
+        }
+    },
+    parseDate: {
+        fn: parseDate,
+        config: {
+            format: { type: 'String' },
+        }
+    },
+    trimStart: {
+        fn: trimStart,
+        config: {
+            char: { type: 'String' }
+        }
+    },
+    trimEnd: {
+        fn: trimEnd,
+        config: {
+            char: { type: 'String' }
+        }
+    },
+    toCamelCase: { fn: toCamelCase, config: {} },
+    toKebabCase: { fn: toKebabCase, config: {} },
+    toPascalCase: { fn: toPascalCase, config: {} },
+    toSnakeCase: { fn: toSnakeCase, config: {} },
+    toTitleCase: { fn: toTitleCase, config: {} },
+};
+
+export function toString(input: any) {
+    return ts.toString(input);
+}
+
+export function toBoolean(input: any) {
+    return ts.toBoolean(input);
+}
+
+export function toUpperCase(input: string) {
+    if (!isString(input)) throw new Error('Input must be a string');
+    return input.toUpperCase();
+}
+
+export function toLowerCase(input: string) {
+    if (!isString(input)) throw new Error('Input must be a string');
+    return input.toLowerCase();
+}
+
+export function trim(input: string, args?: { char: string }) {
+    const char = args ? args.char : ' ';
+    return trimEnd(trimStart(input, { char }), { char });
+}
+
+export function trimStart(input: string, args?: { char: string }): string {
+    if (!isString(input)) throw new Error('Input must be a string');
+    if (args?.char && !isString(args.char)) throw new Error('Input must be a string');
+
+    return ts.trimStart(input, args?.char);
+}
+
+export function trimEnd(input: string, args?: { char: string }): string {
+    if (!isString(input)) throw new Error('Input must be a string');
+    if (args?.char && !isString(args.char)) throw new Error('Input must be a string');
+
+    return ts.trimEnd(input, args?.char);
+}
+
+export function truncate(input: string, args: { size: number }) {
+    if (!isString(input)) throw new Error('Input must be a string');
+    const { size } = args;
+
+    if (!size || !ts.isNumber(size) || size <= 0) throw new Error('Invalid size paramter for truncate');
+    return input.slice(0, size);
+}
+
+export function toISDN(input: any) {
+    let testNumber = ts.toString(input).trim();
+    if (testNumber.charAt(0) === '0') testNumber = testNumber.slice(1);
+
+    // needs to start with a +
+    if (testNumber.charAt(0) !== '+') testNumber = `+${testNumber}`;
+
+    const fullNumber = new PhoneValidator(testNumber).getNumber();
+    if (fullNumber) return String(fullNumber).slice(1);
+
+    throw Error('Could not determine the incoming phone number');
+}
+
+export function toNumber(input: any, args?: { booleanLike?: boolean }) {
+    let result = input;
+
+    if (args?.booleanLike && ts.isBooleanLike(input)) {
+        result = ts.toNumber(toBoolean(result));
+    }
+
+    result = ts.toNumber(result);
+
+    if (isNaN(result)) throw new Error('could not convert to a number');
```

Comment:
We should `Number.isNaN`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

```suggestion
    if (Number.isNaN(result)) throw new Error('could not convert to a number');
```